### PR TITLE
Clang scanner gameplay changes

### PR DESCRIPTION
**Pull request by rmsr**
Things like missing break, unused variables revealing a thinko, etc.

All of these affect gameplay in various ways, I don't think any of them have serious impact.

### CODE REVIEW

**Comment by Hordeking** [comment]
I haven't used clang, so I can't really tease out what this one is about? Is he fixing something so it compiles in clang? Is he just saying he was using clang when he made these changes? @cizra, What do you make of this set of changes? I would think the behavior changes take effect no matter what compiler one uses.

**Comment by rmsr** [comment]
These are defects identified by the clang code sniffer tool: https://clang-analyzer.llvm.org/

**Comment by rmsr** [comment]
To be more specific, clang-scan found various bits of code that failed the sniff test, even if they did compile. These changes attempt to fix spots where (what I perceive to be) the developer's intention was foiled by a thinko or code slippage.

**Comment by rmsr** [comment]
Can I get some reviews on this one? I don't mind if someone NAKs specific commits because they think it distorts gameplay.

File: code/code/cmd/cmd_deathstroke.cc [hunk 1]
```diff
@@ -98,19 +98,20 @@ static int deathstroke(TBeing *caster, TBeing *victim)
...
-    percent = ((10 - (caster->getArmor() / 50)) << 1);
-    percent += caster->getDexReaction() * 5;
-    percent -= victim->getAgiReaction() * 5;
+
+    // with great failure, comes great sorrow *grin*
     // what happens here is that the mob gets a shot at the players vitals
     // ... fair is fair right? 
+    percent = ((10 - (caster->getArmor() / 50)) << 1);
+    percent += victim->getDexReaction() * 5;
+    percent -= caster->getAgiReaction() * 5;
```
Comment (cizra) [inline code/code/cmd/cmd_deathstroke.cc]: Why'd you flip them? Dex is offense, Agi is defense, so it makes sense to check victim's Agi when attacking.

File: lib/txt/news [hunk 2]
```diff
@@ -4,31 +4,18 @@
 07-04-17 : The world command now reports the correct number of distinct
```
Comment (cizra) [inline lib/txt/news]: Why did this file suddenly change so much?

File: code/code/misc/magicutils.cc [hunk 3]
```diff
@@ -1325,10 +1325,7 @@ void TBeing::rawBlind(int level, int duration, saveTypeT save)
   aff.type = SPELL_BLINDNESS;
   aff.bitvector = AFF_BLIND;
   aff.level = level;
-  aff.duration = duration;
-
-  if (save)
-    duration /= 2;
+  aff.duration /= (save ? 2 : 1);
```
Comment (cizra) [inline code/code/misc/magicutils.cc]: Eschew obfuscation

File: lib/txt/news (hunk 2 above)
Comment (rmsr) [reply lib/txt/news]: cleaned up date formatting, out of order entries, indentation, and trailing whitespace.

Comment (rmsr) [reply lib/txt/news]: honestly I can't believe it still applies - we need more news entries for player-visible changes

File: code/code/cmd/cmd_deathstroke.cc (hunk 1 above)
Comment (rmsr) [reply code/code/cmd/cmd_deathstroke.cc]: this calculation is for the counterattack, where the "victim" is the attacker. the previous code was a copy-paste error.

Comment (cizra) [reply code/code/cmd/cmd_deathstroke.cc]: Ohhh right!

**Comment by rmsr** [comment]
De-obfuscated the magicutils comment, should be good to go yes?

**Pull request merged**